### PR TITLE
e2e: allow redefining pull secret location

### PR DESCRIPTION
**Pull request by vrutkovs**
Customize a few parts specific to CI so that the script could be used for local testing

### CODE REVIEW

**Comment by vrutkovs** [comment]
sidevar OOM

/retest

File: dist/Dockerfile.e2e/Dockerfile [hunk 1]
```diff
@@ -24,6 +24,6 @@ ENV PATH="${PATH}:${HOME}/bin"
 
 COPY --from=builder /opt/cincinnati/bin/cincinnati-e2e-test /usr/bin
 COPY --from=builder hack/e2e.sh hack/
-COPY --from=builder dist/openshift/cincinnati.yaml template/
+COPY --from=builder dist/openshift/cincinnati.yaml dist/openshift/
```
Comment (steveej) [inline dist/Dockerfile.e2e/Dockerfile:27]: This change confuses me. Is there a change on CI to make this work as well?

**Review by steveej — COMMENTED** [review]
I can see this being helpful, thanks!

When you write "local testIng", you mean against a cluster not run by CI, right? 

Just as a sidenote and nothing to block on for now, it would be nice if we had a way to test this without a full openshift cluster, e.g. if we could deploy the pod using podman.

File: dist/Dockerfile.e2e/Dockerfile (hunk 1 above)
Comment (vrutkovs) [reply dist/Dockerfile.e2e/Dockerfile:27]: This line needs to be coupled with https://github.com/openshift/cincinnati/pull/244/files#diff-9dac1c15b19c2b8ae927c1c5b610307aR48

Comment (steveej) [reply dist/Dockerfile.e2e/Dockerfile:27]: Ah understood, thanks!

File: hack/e2e.sh [hunk 2]
```diff
@@ -7,10 +7,25 @@
 
 # Prerequirements:
 #   * pull secret (with registry.svc.ci.openshift.org) part in `/tmp/cluster/pull-secret`
+#   * CINCINNATI_IMAGE (optional) - image with graph-builder and policy-engine
 #   * env var IMAGE_FORMAT (e.g `registry.svc.ci.openshift.org/ci-op-ish8m5dt/stable:${component}`)
```
Comment (steveej) [inline hack/e2e.sh:11]: Are we using this anywhere? If not I suggest removing it now which will make the script a bit simpler.

Comment (vrutkovs) [reply hack/e2e.sh:11]: `IMAGE_FORMAT` is passed by our CI, it is used to compose 
a pullspec to the deployment image. In dev builds users would override that with `CINCINNATI_IMAGE` - these sadly cannot be reconciled as the var names are hardcoded in cluster-launch templates on CI

File: hack/e2e.sh [hunk 3]
```diff
@@ -32,7 +35,7 @@ oc project cincinnati-e2e
 oc create secret generic cincinnati-credentials --from-literal=""
 
 # Use this pull secret to fetch images from CI
-oc create secret generic ci-pull-secret --from-file=.dockercfg=/tmp/cluster/pull-secret --type=kubernetes.io/dockercfg
+oc create secret generic ci-pull-secret --from-file=.dockercfg=${PULL_SECRET} --type=kubernetes.io/dockercfg
```
Comment (LalatenduMohanty) [inline hack/e2e.sh:37]: :+1:

File: hack/e2e.sh [hunk 4]
```diff
@@ -27,6 +25,7 @@ echo "IMAGE_TAG=${IMAGE_TAG}"
 # Use defined PULL_SECRET or fall back to CI location
 PULL_SECRET=${PULL_SECRET:-/tmp/cluster/pull-secret}
 
+set -euo pipefail
```
Comment (LalatenduMohanty) [inline hack/e2e.sh:28]: :+1:

**Review by steveej — APPROVED** [review]
/lgtm
/hold cancel

**Pull request merged**